### ISSUE TITLE
python3.11 pip install failing

### ISSUE THREAD

**Issue by rhubert** [BobBuildTool/bob#513]
```
$ python3 -m pip install --user BobBuildTool

 [....]


      running build_sphinx
      /usr/local/lib/python3.11/site-packages/setuptools/_distutils/dist.py:987: RemovedInSphinx70Warning: setup.py build_sphinx is deprecated.
        cmd_obj.run()
      Running Sphinx v6.1.3

      Extension error:
      Could not import extension sphinxcontrib.applehelp (exception: No module named 'sphinxcontrib.applehelp')
      [end of output]
```
complete log: [log.txt](https://github.com/BobBuildTool/bob/files/10987812/log.txt)

Manually installing `sphinxcontrib.applehelp` did not solve this :/ Any ideas?

**Comment by jkloetzke** [comment]
It seems to help to install Sphinx completely before installing Bob:

    pip install sphinx
    pip install BobBuildTool

I'm not yet sure what the real root cause is. In principle sphinx is just a build dependency and does not need to be installed. Strange...

**Comment by kolewu** [comment]
You only need to install sphinx if there is no wheel for the python version used. 0.22.0 is currently only available for python 3.6 -- 3.10. So with 3.11 you need sphinx. And the current version 7 of sphinx removed the support for `setup.py`.

So you now have to install `sphinx==6.2.1` before installing bob.
